packages/backend/src/errors/step.ts

```diff
@@ -0,0 +1,9 @@
+import { IStepError } from '@plumber/types'
+
+export default class StepError extends Error {
+  constructor(error: IStepError, options?: unknown) {
+    const computedMessage = JSON.stringify(error)
```

Comment:
nit: actually looking at this more closely... i think could be useful to expose the initial constructor params inside the object.

e.g.
```
export default class StepError extends Error {
  // can call this something different to make this more understandable
  params: IStepError

  constructor(error: IStepError, options?: unknown) {
    // also thinking we don't need to stringify here if we have params availiable.
    super(error.name, options)
    this.name = this.constructor.name
    this.params = error
  }
}
```

Reply:
nvm I see why you need to stringify the contents, it's for the front end. then nvm ignore this